File: backend/app/schemas/analysis.py
from pydantic import BaseModel


class ClassificationResult(BaseModel):
    category_id: int | None
    category_name: str
    confidence: float
    explanation: str


class PriorityResult(BaseModel):
    priority_id: int | None
    priority_name: str
    confidence: float
    explanation: str


class SimilarArticle(BaseModel):
    id: int
    title: str
    excerpt: str
    category_id: int | None
    score: float


class GeneratedAnswer(BaseModel):
    response_text: str
    model_name: str
    provider_name: str
    sources_used: str | None


class AnalysisResult(BaseModel):
    ticket_id: int
    classification: ClassificationResult
    priority: PriorityResult
    similar_articles: list[SimilarArticle]
    ai_response: GeneratedAnswer

File: frontend/src/types/analysis.ts
export interface ClassificationResult {
  category_id: number | null
  category_name: string
  confidence: number
  explanation: string
}

export interface PriorityResult {
  priority_id: number | null
  priority_name: string
  confidence: number
  explanation: string
}

export interface SimilarArticle {
  id: number
  title: string
  excerpt: string
  category_id: number | null
  score: number
}

export interface GeneratedAnswer {
  response_text: string
  model_name: string
  provider_name: string
  sources_used: string | null
}

export interface AnalysisResult {
  ticket_id: number
  classification: ClassificationResult
  priority: PriorityResult
  similar_articles: SimilarArticle[]
  ai_response: GeneratedAnswer
}
